compiler/modules/front.syntax/token.bal

```diff
@@ -236,12 +236,14 @@ class Tokenizer {
         }
     }
 
+    // peeks the next token. considers qualified identifier as a single token, ie if current token
```

Comment:
+1 for having a parameter such as `skipQualIdent` that control the behavior.

But couldn't find an nice way to return the additional tokens, I tried

```ballerina
type Peekable VariableTokenCode|FixedToken;
function peek() returns Peekable|[Peekable, Peekable, Peekable...] { ... }
```
Single Peekable is returned as is. if more than one is returned, it's a list. This is ugly because other side need to an additional check.


```ballerina
type Peekable VariableTokenCode|FixedToken;
function peek() returns [Peekable, Peekable...] { ... }
```
This look nicer, but there is unnecessary wrapping going on.

We are only using `peek` in two places, need for this is not concrete just yet. Shall we stick to the parameter for now?

Reply:
OK, but add a comment to make it clear what the semantic of `peek` are.